File: GeneratedEntityReaderClasses/Classes/ValueRange.cs
namespace GeneratedNoitaClasses;
public class ValueRange :  iNoitaType<ValueRange>{
	public NoitaFloat min { get; set; } = new();
	public NoitaFloat max { get; set; } = new();

    public void Read(NoitaStream s){
		min.Read(s);
		max.Read(s);

    }
    public void Write(NoitaStream s){
		min.Write(s);
		max.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/ValueRange.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ValueRange:
    min:NoitaFloat = field(default_factory=NoitaFloat)
    max:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.min.Read(s)
        self.max.Read(s)
    
    def Write(self, s:NoitaStream):
        self.min.Write(s)
        self.max.Write(s)

